pkg/util/winutil/winstrings.go

```diff
@@ -8,69 +8,69 @@
 package winutil
 
 import (
-	"bytes"
+	"unicode/utf16"
+	"unsafe"
 )
 
 // ConvertWindowsStringList Converts a windows-style C list of strings
 // (single null terminated elements
 // double-null indicates the end of the list) to an array of Go strings
 func ConvertWindowsStringList(winput []uint16) []string {
-	var retstrings []string
-	var buffer bytes.Buffer
 
-	for i := 0; i < (len(winput) - 1); i++ {
-		if winput[i] == 0 {
-			retstrings = append(retstrings, buffer.String())
-			buffer.Reset()
-
-			if winput[i+1] == 0 {
-				return retstrings
-			}
-			continue
+	if len(winput) < 2 {
+		return []string{}
+	}
+	if winput[len(winput)-1] == 0 {
+		winput = winput[:len(winput)-1] // remove terminating null
+	}
+	val := make([]string, 0, 5)
+	from := 0
+	for i, c := range winput {
+		if c == 0 {
```

Comment:
If the double NULL, indicating the end of the list, occurs anywhere but the end of the slice, this code will not handle it properly. The previous implementation did though. I'd suggest following its structure, and just replacing the string decoding parts.